docs/source/background/scf_notes.md

```diff
@@ -0,0 +1,86 @@
+<!--
+  ~ Copyright 2025 NWChemEx-Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+-->
+
+# SCF Procedure
+
+## SCF Driver
+
+1. Obtain the AO parameters as well as the chemical system (inputs)
+2. Assign the AO parameters as atomic orbitals
+3. Create the Hamitonian from the Chemical System (Submod "Hamiltonian" used, currently only the BO Approximation from NUX)
+    1. For the BO Approximation Hamiltonian:
+        1. From the Chemical System, the Electrons and Nuclei are used to form the Hamiltonian
+        2. Each electron gets a 1 electron kinetic energy term
+        3. Each nuclei gets an Electron-Nuclei potential energy term
+        4. If more than 1 electron, an Electron-Electron potential energy repulson term is added
+        5. If more than 1 nuclei, a Nuclei-Nuclei potential energy repulsion term is added
+        6. In order, the Electron kinetic energy, Electron-nucleus attraction energy, and Electron-electron repulsion energy are
+           formed to create the Electronic Hamiltonian
+        7. The Nuclear-nuclear repulsion energy is also created
+        8. The BO Approximation returns the constructed Hamiltonian, H.
+4. With the guess submod, (of type InitialGuess), the inputs are the Hamiltonian and the AOs, and the return value is the
+   initial wavefunction
+    1. The main module used is the "Core guess" module, which first builds the fock operator with 0 density
+    2. I am not sure on what happens during lines 78-92:
+        ```cpp
+        simde::type::cmos cmos(tensor{}, aos, tensor{});
+        NElectronCounter visitor;
+        H.visit(visitor);
+        auto n_electrons = visitor.n_electrons;
+        if(n_electrons % 2 != 0)
+            throw std::runtime_error("Assumed even number of electrons");
+
+        typename rscf_wf::orbital_index_set_type occs;
+        using value_type = typename rscf_wf::orbital_index_set_type::value_type;
+        for(value_type i = 0; i < n_electrons / 2; ++i) occs.insert(i);
+
+        rscf_wf zero_guess(occs, cmos);
+        auto& update_mod = submods.at("Guess updater");
+        const auto& Psi0 = update_mod.run_as<update_pt>(f, zero_guess);```
+
+    3. From there, Psi0 is returned
+
+### SCF Loop
+
+5. The work horse of the module, the submod "Optimizer", uses `scf_loop.cpp` and the module `Loop` to optimize
```

Comment:
I like how you separated the loop from the driver. I'd finish out the driver section first though, by adding a step that says something like "the SCF energy is then optimized with respect to the wavefunction by calling a module of property type `Optimize<Types,Go,Here>`.